components/org.wso2.carbon.security.sts.common/src/main/java/org/wso2/carbon/identity/sts/common/identity/provider/AttributeCallbackHandler.java

```diff
@@ -109,6 +111,10 @@ public void handle(SAMLCallback callback) throws SAMLException {
                 splitArr = userIdentifier.split(",")[0].split("=");
                 if (splitArr.length == 2) {
                     userIdentifier = splitArr[1];
+                } else if (!userIdentifier.contains(UserCoreConstants.DOMAIN_SEPARATOR)) {
+                    // if the user identifier is not qualified with user store domain, making it so.
```

Comment:
In the current implementation, the value retrieved for the `userIdentifier` contains the user store domain since the user store domain is used in things like populating the claim values.

However, with the change [[1](https://github.com/nipunsampath/identity-inbound-auth-sts/blob/dc39fc824d55ad6d0e91d51ea7dbe8ea1f4713db/components/org.wso2.carbon.security.sts.common/src/main/java/org/wso2/carbon/identity/sts/common/util/ServicePasswordCallbackHandler.java#L257)] added in this PR, if we configure IS not to include the user store domain in the local subject identifier, the user store domain will not be included in the `userIdentifier`. That's why this change has been added and essentially it's resolving an incompatibility.

[1] https://github.com/nipunsampath/identity-inbound-auth-sts/blob/dc39fc824d55ad6d0e91d51ea7dbe8ea1f4713db/components/org.wso2.carbon.security.sts.common/src/main/java/org/wso2/carbon/identity/sts/common/util/ServicePasswordCallbackHandler.java#L257